src/app/shared/components/atoms/input-text/input-text.component.ts

```diff
@@ -25,7 +38,32 @@ export class InputTextComponent implements OnInit {
   }
 
   change(event: Event): void {
+    this.markAsTouched();
     const value = (event.target as any)?.value;
     this.valueChange.emit(value);
+    this.onChange(value);
+  }
+
+  writeValue(value: string): void {
+    this.value = value;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
```

Comment:
These will, at the very least, be functions.

```suggestion
  registerOnChange(fn: () => void)): void {
    this.onChange = fn;
  }

  registerOnTouched(fn: () => void)): void {
```